borg/testsuite/archiver.py

```diff
@@ -1111,12 +1160,201 @@ def test_debug_put_get_delete_obj(self):
+        ]:
+            self.setUp()
+            try:
+                self._test_rewrite_interrupt(create_stale_temp, change_args, interrupt_early)
+            finally:
+                self.tearDown()
+        self.setUp()
+
+    def _test_rewrite_chunker_interrupt_patch(self):
+        real_add_chunk = Cache.add_chunk
+
+        def add_chunk(*args, **kwargs):
+            frame = inspect.stack()[2]
+            try:
+                caller_self = frame[0].f_locals['self']
+                caller_self.interrupt = True
+            finally:
+                del frame
+            return real_add_chunk(*args, **kwargs)
+        return add_chunk
+
+    def test_rewrite_rechunkify_interrupt(self):
+        self.create_regular_file('file1', size=1024 * 80)
+        self.cmd('init', self.repository_location)
+        self.cmd('create', self.repository_location + '::test', 'input')
+        archive_before = self.cmd('list', self.repository_location + '::test', '--format', '{sha512}')
+        with patch.object(Cache, 'add_chunk', self._test_rewrite_chunker_interrupt_patch()):
+            self.cmd('rewrite', '-p', '--chunker-params', '16,18,17,4095', self.repository_location)
+        assert 'test.rewrite' in self.cmd('list', self.repository_location)
+        output = self.cmd('rewrite', '-svp', '--debug', '--chunker-params', '16,18,17,4095', self.repository_location)
+        assert 'Found test.rewrite, will resume' in output
+        assert 'Copied 1 chunks from a partially processed item' in output
+        archive_after = self.cmd('list', self.repository_location + '::test', '--format', '{sha512}')
+        assert archive_after == archive_before
+
+    def test_rewrite_changed_source(self):
+        self.create_test_files()
+        self.cmd('init', self.repository_location)
+        self.cmd('create', self.repository_location + '::test', 'input')
+        with patch.object(ArchiveRewriter, 'process_item', self._rewrite_interrupt_patch(1)):
+            self.cmd('rewrite', self.repository_location, 'input/dir2')
+        assert 'test.rewrite' in self.cmd('list', self.repository_location)
+        self.cmd('delete', self.repository_location + '::test')
+        self.cmd('create', self.repository_location + '::test', 'input')
+        output = self.cmd('rewrite', self.repository_location, 'input/dir2')
+        assert 'Source archive changed, will discard test.rewrite and start over' in output
+
+    def test_rewrite_refuses_temporary(self):
+        self.cmd('init', self.repository_location)
+        self.cmd('rewrite', self.repository_location + '::cba.rewrite', exit_code=2)
+        self.cmd('rewrite', self.repository_location + '::abc.rewrite.temp', exit_code=2)
+
 
 @unittest.skipUnless('binary' in BORG_EXES, 'no borg.exe available')
 class ArchiverTestCaseBinary(ArchiverTestCase):
     EXE = 'borg.exe'
     FORK_DEFAULT = True
 
+    @unittest.skip('patches objects')
+    def test_rewrite_rechunkify_interrupt(self):
+        ...
```

Comment:
More like a `... there is a way we can somehow test this, and we probably should`. I haven't looked into that, though. If it's to complicated to test I'd put a `pass` in there.